common/src/stack/config-server/cgi/index.cgi

```diff
@@ -15,16 +14,16 @@ out += '<html>'
 out += '<body>'
 out += '<table>'
 
-listing = os.listdir(dir)
+listing = os.listdir(directory)
 listing.sort(key=str.lower)
-for file in listing:
-	if file not in [ 'index.cgi' ]:
+for filename in listing:
+	if 'index.cgi' not in filename:
 		out += '<tr><td>\n'
 
-		if os.path.isdir(os.path.join(dir, file)):
-			out += '<a href="%s/">%s/</a>\n' % (file, file)
+		if os.path.isdir(os.path.join(directory, filename)):
+			out += '<a href="%s/">%s/</a>\n' % (filename, filename)
```

Comment:
Actually, the Python optimization for "+=" on strings in a little more nuanced.

If the left hand side (the string being added onto) has a single reference (EG: is local to the function scope), then Python has an optimization that grow the string under the hood instead of creating a branch new string and copying both pieces into it.

If the string isn't local (it's a member variable, etc) then the "+=" behavior allocates memory for the combined string length and then copies both strings into the new one. 

That is slow, but you would only notice with a lot of string concats going on.